ml_tools/eolearn/tests/test_train_set_mask.py

```diff
@@ -0,0 +1,58 @@
+import unittest
+import numpy as np
+
+from eolearn.core import FeatureType, EOPatch
+from eolearn.ml_tools import TrainSetMaskTask
+
+
+class TestTrainSet(unittest.TestCase):
+
+    def test_train_set_mask(self):
+        new_name = 'TEST_TRAIN_MASK'
+
+        input_mask_feature = (FeatureType.MASK_TIMELESS, 'TEST')
+        input_label_feature = (FeatureType.LABEL_TIMELESS, 'TEST')
+        new_mask_feature = (FeatureType.MASK_TIMELESS, new_name)
+        new_label_feature = (FeatureType.LABEL_TIMELESS, new_name)
+
+        self.assertRaises(ValueError, TrainSetMaskTask, input_mask_feature, None)
+        self.assertRaises(ValueError, TrainSetMaskTask, input_mask_feature, 1.5)
+        self.assertRaises(ValueError, TrainSetMaskTask, input_mask_feature, [0.5, 0.3, 0.7])
+
+        shape = (8, 8, 3)
+        size = np.prod(shape)
+
+        data = np.ones(shape=shape, dtype=np.int)
+
+        data[0:2, 0:2, :] = 11
+        data[0:2, 2:4, :] = 22
+        data[2:4, 0:2, :] = 33
+        data[2:4, 2:4, :] = 44
+        data[0:4, 4:8, :] = 55
+        data[4:8, 0:4, :] = 66
+        data[4:8, 4:8, :] = 77
+
+        patch = EOPatch()
+        patch[(FeatureType.MASK_TIMELESS, 'TEST')] = data
+        patch[(FeatureType.LABEL_TIMELESS, 'TEST')] = data.copy().reshape((size,))
+
+        bins = [0.2, 0.5, 0.8]
+
+        patch = TrainSetMaskTask((*input_mask_feature, new_name), bins)(patch, seed=1)
+        self.assertTrue(set(np.unique(patch[new_mask_feature])) <= set(range(len(bins) + 1)))
```

Comment:
Above you could define
```Python
interval_list = [(0, 2, 0, 2), (0, 2, 2, 4) ,...]
```
use this indices from such list to set `data[i1: i2, j1: j2, :] = 11 * index`.
Then  you could use these indices again here to make sure that each of these slices has a unique value